File: mysite/groundmenu_spa/store_information.py
import json
#from channels.generic.websocket import WebsocketConsumer
from channels.generic.websocket import AsyncWebsocketConsumer
#from asgiref.sync import async_to_sync  # async_to_sync() : 非同期関数を同期的に実行する際に使用する。
from channels.db import database_sync_to_async
from .models import Store
from django.core.serializers import serialize

USERNAME_SYSTEM = '*system*'

# ChatConsumerクラス: WebSocketからの受け取ったものを処理するクラス
class store_information( AsyncWebsocketConsumer ):

    # WebSocket接続時の処理
    async def connect( self ):
        await self.accept()

    # WebSocket切断時の処理
    async def disconnect( self, close_code ):
        return

    async def receive( self, text_data ):
        json_data = json.loads( text_data )
        action = json_data.get( 'action' )
        if( 'create' == action ):
            print('create')
        elif( 'read' == action ):
            store_information = await self.store_read(json_data)
        elif( 'update' == action ):
            store_information = await self.store_update(json_data)
        elif( 'delete' == action ):
            print('delete')
        else:
            return
        store_information = json.dumps(store_information)
        await self.send( store_information )

    @database_sync_to_async
    def store_update(self, json_data):
        store_id = json_data.get('store_id')
        store_name = json_data.get('store_name')
        seating_capacity = json_data.get('seating_capacity')
        takeout_support = json_data.get('takeout_support')
        raw_store_data = Store.objects.filter(store_id = store_id)
        raw_store_data.update(store_name = store_name, seating_capacity = seating_capacity, takeout_support = takeout_support)
        raw_store_data = serialize('json', (raw_store_data))
        raw_store_data = json.loads(raw_store_data)
        store_information = self.store_information_query_to_json(raw_store_data)
        return store_information

    @database_sync_to_async
    def store_read(self, json_data):
        store_id = json_data.get('store_id')
        raw_store_data = Store.objects.filter(store_id = store_id)
        raw_store_data = serialize('json', (raw_store_data))
        raw_store_data = json.loads(raw_store_data)
        store_information = self.store_information_query_to_json(raw_store_data)
        return store_information

    def store_information_query_to_json(self, raw_store_data):
        str_store_data = raw_store_data[0]['fields']
        store_id = raw_store_data[0]['pk']
        store_name = str_store_data['store_name']
        seating_capacity = str_store_data['seating_capacity']
        takeout_support = str_store_data['takeout_support']
        json_data = {
            'store_id': store_id,
            'store_name': store_name,
            'seating_capacity': seating_capacity,
            'takeout_support': takeout_support,
        }

        return json_data
         
        

File: vue3/src/store/modules/storeInformation.js
const state = {
    storeID: '',
    storeName: '',
    seatingCapacity: '',
    takeoutSupport: '',
};

const getters = {
    storeID: state => state.storeID,
    storeName: state => state.storeName,
    seatingCapacity: state => state.seatingCapacity,
    takeoutSupport: state => state.takeoutSupport,
};

const mutations = {
    storeInformation_update(state, newStoreInformation){
        state.storeID = newStoreInformation.store_id
        state.storeName = newStoreInformation.store_name
        state.seatingCapacity = newStoreInformation.seating_capacity
        state.takeoutSupport = newStoreInformation.takeout_support
    },
};

const actions ={
    storeInformation_read({commit}, storeInformation){
        commit('storeInformation_update', storeInformation)
    },
    storeInformation_update({ commit }, storeInformation){
        const new_store_information = JSON.stringify({
            'action':'update',
            'store_id': storeInformation.store_id,
            'store_name': storeInformation.store_name,
            'seating_capacity': storeInformation.seating_capacity,
            'takeout_support': storeInformation.takeout_support,
        });
        this.$store_information.send( new_store_information );
        commit("storeInformation_update", storeInformation)
    },
    // seatingCapacityChange({ commit }, newSeatingCapacity){
    //     commit("seatingCapacityChange", newSeatingCapacity)
    // },
    // takeoutSupportChange({ commit }, newTakeoutSupport){
    //     commit("takeoutSupportChange", newTakeoutSupport)
    // }
};



export default {
    state,
    getters,
    mutations,
    actions,
}